swaybar/render.c

```diff
@@ -23,8 +23,32 @@ static const int WS_HORIZONTAL_PADDING = 5;
 static const double WS_VERTICAL_PADDING = 1.5;
 static const double BORDER_WIDTH = 1;
 
-static uint32_t render_status_line_error(cairo_t *cairo,
-		struct swaybar_output *output, double *x) {
+struct render_call {
```

Comment:
Given that this is replacing `cairo_t *cairo` in most places, perhaps `struct render_context`/`cairo_context`/`cairo_state` (or similar) is a better name?